server/core_tournament.go

```diff
@@ -222,7 +217,26 @@ WHERE duration > 0 AND start_time >= $1 AND end_time <= $2 AND category >= $3 AN
 		params = append(params, 127)
 	}
 
-	// To ensure that there are more records, so the cursor is returned/
+	if startTime >= 0 {
+		params = append(params, time.Unix(int64(startTime), 0).UTC())
+	} else {
+		params = append(params, time.Unix(0, 0).UTC())
+	}
+
+	if endTime == 0 {
+		query += " AND end_time = $4"
+		params = append(params, time.Unix(0, 0).UTC())
+	} else if int64(endTime) < now.Unix() {
+		// if end time is set explicitly in the past
+		query += " AND end_time <= $4"
+		params = append(params, time.Unix(int64(endTime), 0).UTC())
+	} else {
+		// if end time is in the future, return both tournaments that end in the future as well as the ones that never end.
+		query += " AND (end_time <= $4 OR end_time = '1970-01-01 00:00:00 UTC')"
```

Comment:
Open question:

Should I update the query to something like this:

```
query += " AND ((end_time >= now() AND end_time <= $4) OR end_time = '1970-01-01 00:00:00 UTC')"
```

This would mean any end_time between now, and in the future - excluding tournaments that ended previously. I don't think this is needed because we are doing that check in Nakama anyways.

Reply:
I think it's fine as it is.